### PR TITLE
Add ability to use concatenation instead of addition in FPN

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
Hi!
See: https://github.com/qubvel/segmentation_models.pytorch/issues/78

### CODE REVIEW

**Comment by ternaus** [comment]
```
if self.merge_policy == 'add':
    x = s5 + s4 + s3 + s2
elif self.merge_policy == 'cat':
    x = torch.cat([s5, s4, s3, s2])
```
=>

```
if self.merge_policy == 'add':
    x = s5 + s4 + s3 + s2
elif self.merge_policy == 'cat':
    x = torch.cat([s5, s4, s3, s2])
else:
   raise ValueError("Only add and merge policies supported, but got {}".format(self.merge_policy))
```

File: segmentation_models_pytorch/fpn/model.py [hunk 1]
```diff
@@ -36,7 +37,11 @@ def __init__(
             dropout=0.2,
             activation='sigmoid',
             final_upsampling=4,
+            decoder_merge_policy='add'
     ):
+        if decoder_merge_policy not in ['add', 'cat']:
+            raise ValueError("`merge_policy` must be one of: ['add', 'cat']")
+
```
Comment (qubvel) [inline segmentation_models_pytorch/fpn/model.py]: I think this is responibility of decoder to know what kind of merge policy to use, however in this case we first build encoder and load its weights which is time and memory consuming operation.. what do you think? is it it better to move this to decoder?

File: segmentation_models_pytorch/fpn/decoder.py [hunk 2]
```diff
@@ -100,7 +103,10 @@ def forward(self, x):
         s3 = self.s3(p3)
         s2 = self.s2(p2)
 
-        x = s5 + s4 + s3 + s2
+        if self.merge_policy == 'add':
+            x = s5 + s4 + s3 + s2
+        elif self.merge_policy == 'cat':
+            x = torch.cat([s5, s4, s3, s2])
```
Comment (qubvel) [inline segmentation_models_pytorch/fpn/decoder.py]: In case of 'cat' `final_conv` should have segmentation_channels * 4 input channels (line 89)
And set `dim=1` for cat (deafult is 0, which is wrong).

File: segmentation_models_pytorch/fpn/model.py (hunk 1 above)
Comment (SiarheiFedartsou) [reply segmentation_models_pytorch/fpn/model.py]: > however in this case we first build encoder and load its weights which is time and memory consuming 

Hm, you mean that it can be annoying for library user to know that he passed wrong merge_policy after waiting for 20 minutes for weights downloading? :) What do you think if move this check to the utility function and will do such checks in both decoder and model?

Comment (SiarheiFedartsou) [reply segmentation_models_pytorch/fpn/model.py]: Just added `check_merge_policy` function. Please see and let me know what do you think about such approach.

File: segmentation_models_pytorch/fpn/decoder.py (hunk 2 above)
Comment (SiarheiFedartsou) [reply segmentation_models_pytorch/fpn/decoder.py]: Fixed, thanks.

File: segmentation_models_pytorch/fpn/model.py [hunk 3]
```diff
@@ -49,6 +52,7 @@ def __init__(
             final_channels=classes,
             dropout=dropout,
             final_upsampling=final_upsampling,
+            merge_policy=check_merge_policy(decoder_merge_policy)
```
Comment (qubvel) [inline segmentation_models_pytorch/fpn/model.py]: Lets make it only inside decoder ;)
without utility function, just 
```python
    if merge_policy not in ['add', 'cat']:
        raise ValueError("`merge_policy` must be one of: ['add', 'cat'], got {}".format(merge_policy)
```
And without `f` string formatting (in this project it is not used, it is better to be cosistent)

Comment (SiarheiFedartsou) [reply segmentation_models_pytorch/fpn/model.py]: Fixed.

**Comment by qubvel** [comment]
Thanks a lot!
